galpy/potential/SCFPotential.py

```diff
@@ -685,6 +714,108 @@ def integrand(xi, costheta):
         Acos[:,:,0] = 2*I**-1 * integrated*constants
         
         return Acos, Asin
+    
+def scf_compute_coeffs_nbody(pos,mass,N,L,a=1., radial_order=None, costheta_order=None, phi_order=None):
+        
+        """        
+        NAME:
+
+           scf_compute_coeffs
+
+        PURPOSE:
+
+           Numerically compute the expansion coefficients for a given triaxial density
+
+        INPUT:
+
+           pos - Positions of particles
+           
+           m - mass of particles
+
+           N - size of the Nth dimension of the expansion coefficients
+
+           L - size of the Lth and Mth dimension of the expansion coefficients
+           
+           a - parameter used to shift the basis functions
+
+           radial_order - Number of sample points of the radial integral. If None, radial_order=max(20, N + 3/2L + 1)
+
+           costheta_order - Number of sample points of the costheta integral. If None, If costheta_order=max(20, L + 1)
+
+           phi_order - Number of sample points of the phi integral. If None, If costheta_order=max(20, L + 1)
+
+        OUTPUT:
+
+           (Acos,Asin) - Expansion coefficients for density dens that can be given to SCFPotential.__init__
+
+        HISTORY:
+
+           2020-11-18 - Written - Morgan Bennett
+
+        """ 
+        
+        n = numpy.arange(0,N)
+        l = numpy.arange(0,L)
+        m = numpy.arange(0,L)
+
+        r = numpy.sqrt(pos[0]**2+pos[1]**2+pos[2]**2)
+        phi = numpy.arctan2(pos[1],pos[0])
+        costheta = pos[2]/r
```

Comment:
Astropy does not (yet) support the ufunc ``lpmv``, so dimensionless quantities need to be recast.
The following works on both normal and astropy Quantity arrays.

``costheta = u.Quantity(pos[2]/r, copy=False).to_value(u.one)``

Reply:
To deal with positions given with units, we'll just have to apply the relevant decorator. I think probably the one used for the 2D potential functions should work?